### PR TITLE
Add snap package.

### PR DESCRIPTION
**Pull request by soumyaDghosh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,9 +60,11 @@ Options:
 
 Option 1 (recommended): Install the flatpak from [Flathub][flathub]: `flatpak --user install com.adilhanney.saber`.
 
-Option 2: Download `Saber.flatpak` from the latest [Release][releases] and install with `flatpak --user install Saber.flatpak`
+Option 2: Install the snap from <a href="https://snapcraft.io/saber">Snap-Store</a>
```
Comment (adil192) [inline README.md:63]: Please can you make the snap Option 4 since I won't be maintaining it?
Also can you add that it's unofficially maintained

**Review by adil192 — CHANGES_REQUESTED** [review]

**Comment by adil192** [comment]
https://github.com/soumyaDghosh/saber#remaining-tasks

Please complete the remaining tasks, including Snapcrafters testing and getting the repo forked into github.com/snapcrafters

**Comment by adil192** [comment]
[`saber/snap/local/saber.desktop`](https://github.com/soumyaDghosh/saber/blob/d70b01ffa8a18f1a215c595d842e61d05e31dc91/snap/local/saber.desktop#L2)
```desktop
[Desktop Entry]
Version=0.8.1
Type=Application

Name=Saber
Comment=A cross-platform libre handwritten notes app.
Categories=Office;WordProcessor;

Icon=icon.png
Exec=saber
Terminal=false
StartupWMClass=saber
```

I believe the `Version` here should be the version of the `.desktop` file spec used, e.g. `1.0` like [`flatpak/com.adilhanney.saber.desktop`](https://github.com/adil192/saber/blob/main/flatpak/com.adilhanney.saber.desktop)

**Comment by adil192** [comment]
[`snap/snapcraft.yaml`](https://github.com/soumyaDghosh/saber/blob/d70b01ffa8a18f1a215c595d842e61d05e31dc91/snap/snapcraft.yaml#L29)

```yaml
  saber:
    # See 'snapcraft plugins'
    plugin: flutter
    source: https://github.com/adil192/saber.git
    source-tag: v${SNAPCRAFT_PROJECT_VERSION}
    build-packages:
      - libsecret-1-dev 
      - libjsoncpp-dev
    stage-packages:
      - libjsoncpp1
```

Please add in the missing dependency from [flutter_secure_storage](https://pub.dev/packages/flutter_secure_storage)'s example:

```yaml
parts:
  uet-lms:
    source: .
    plugin: flutter
    flutter-target: lib/main.dart
    build-packages:
      - libsecret-1-dev
      - libjsoncpp-dev
    stage-packages:
      - libsecret-1-0
      - libjsoncpp-dev
```

Also please add in `flutter-target: lib/main_snap.dart`

**Comment by adil192** [comment]
[`snap/snapcraft.yaml`](https://github.com/soumyaDghosh/saber/blob/d70b01ffa8a18f1a215c595d842e61d05e31dc91/snap/snapcraft.yaml#L45)

```yaml
apps:
  saber:
    command: bin/saber
    extensions: [flutter-master]
    common-id: com.adilhanney.saber
    desktop: saber.desktop
    plugs: 
      - home
```

Please change this to `extensions: [flutter-stable]`. I don't believe Saber will even compile on Flutter master. Have you tested whether the application runs?

File: README.md (hunk 1 above)
Comment (soumyaDghosh) [reply README.md:63]: I will maintain it continiously. It'll anyways need very less manual interuption.

Comment (adil192) [reply README.md:63]: I'd still prefer it as Option 4 here, plus mentioning that it's unofficially maintained

**Pull request closed without merging**